src/ConnectedMode.UnitTests/Suppressions/SuppressionIssueStoreUpdaterTests.cs

```diff
@@ -0,0 +1,295 @@
+using SonarQube.Client.Models;
+using static SonarLint.VisualStudio.TestInfrastructure.NoOpThreadHandler;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Suppressions
+{
+    [TestClass]
+    public class SuppressionIssueStoreUpdaterTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<SuppressionIssueStoreUpdater, ISuppressionIssueStoreUpdater>(
+                MefTestHelpers.CreateExport<ISonarQubeService>(),
+                MefTestHelpers.CreateExport<IServerQueryInfoProvider>(),
+                MefTestHelpers.CreateExport<IServerIssueStoreWriter>(),
+                MefTestHelpers.CreateExport<ILogger>());
+        }
+
+        [TestMethod]
+        [DataRow(null, null)]
+        [DataRow(null, "branch")]
+        [DataRow("projectKey", null)]
+        public async Task UpdateAll_MissingServerQueryInfo_NoOp(string projectKey, string branchName)
+        {
+            // Server query info is not available -> give up
+            var queryInfo = CreateQueryInfoProvider(projectKey, branchName);
+            var server = new Mock<ISonarQubeService>();
+            var writer = new Mock<IServerIssueStoreWriter>();
+
+            var testSubject = CreateTestSubject(queryInfo.Object, server.Object, writer.Object);
+
+            await testSubject.UpdateAllServerSuppressionsAsync();
+
+            queryInfo.Verify(x => x.GetProjectKeyAndBranchAsync(It.IsAny<CancellationToken>()), Times.Once());
+            server.Invocations.Should().HaveCount(0);
+            writer.Invocations.Should().HaveCount(0);
+        }
+
+        [TestMethod]
+        public async Task UpdateAll_HasServerQueryInfo_ServerQueriedAndStoreUpdated()
+        {
+            // Happy path - fetch and update
+            var queryInfo = CreateQueryInfoProvider("project", "branch");
+            
+            var issue = CreateIssue("issue1");
+            var server = CreateSonarQubeService("project", "branch", issue);
+            
+            var writer = new Mock<IServerIssueStoreWriter>();
+
+            var testSubject = CreateTestSubject(queryInfo.Object, server.Object, writer.Object);
+
+            await testSubject.UpdateAllServerSuppressionsAsync();
+
+            queryInfo.Verify(x => x.GetProjectKeyAndBranchAsync(It.IsAny<CancellationToken>()), Times.Once());
+            server.Verify(x => x.GetSuppressedIssuesAsync("project", "branch", It.IsAny<CancellationToken>()), Times.Once());
+            writer.Verify(x => x.AddIssues(new[] { issue }, true), Times.Once);
+
+            server.Invocations.Should().HaveCount(1);
+            writer.Invocations.Should().HaveCount(1);
+        }
+
+        [TestMethod]
+        public async Task UpdateAll_RunOnBackgroundThread()
+        {
+            var callSequence = new List<string>();
+
+            var queryInfo = new Mock<IServerQueryInfoProvider>();
+            var threadHandling = new Mock<IThreadHandling>();
```

Comment:
A more robust check would check that all of the services are called in the expected order, but this is enough for now.